torch/_library/custom_ops.py

```diff
@@ -630,6 +631,24 @@ def backend_select(keyset, *args, **kwargs):
     def __call__(self, *args, **kwargs):
         return self._opoverload(*args, **kwargs)
 
+    def register_vmap(
+        self,
+        func: Callable,
+        /,
+        *,
+        setup_context: Optional[Callable] = None,
+    ) -> None:
+        schema = self._opoverload._schema
+        if not _library.utils.is_functional_schema(schema):
+            raise RuntimeError(
+                f"Cannot register vmap formula for non-functional operator "
+                f"{self} with schema {schema}. Please create "
+                f"a functional operator and register a vmap formula for that."
+            )
```

Comment:
I don't think we need this check, the user can register a vmap rule for in-place operations (unlike the autograd case where we may clobber some inputs that need to be saved for backward)